server/src/main/java/io/crate/action/sql/ResultReceiver.java

```diff
@@ -39,4 +39,10 @@ public interface ResultReceiver<T> extends CompletionListenable<T> {
     void allFinished();
 
     void fail(Throwable t);
+
+    /**
+     * Flushes out pending messages but doesn't send any other additional messages to the client
+     * Doesn't trigger the completion future
+     **/
+    void abort();
```

Comment:
I don't see any of
`channel.writePendingMessages(delayedWrites);`
`channel.flush();`
happening, do I miss something?

Reply:
Added them in a fixup. I'm not sure if it's really needed, tests pass with both.

Reply:
Looks like not needed when using a `ResultSet` as `ResultSetReceiver.batchFinished()` does this on its own. I wonder if this is needed then as no other `ResultReceiver` impl is affected of this?

Reply:
Right.  I guess in that case we can remove the abort and instead call nothing